src/aap_eda/api/views/external_webhook.py

```diff
@@ -0,0 +1,154 @@
+
+    def _update_test_data(
+        self,
+        error_message: str = "",
+        content_type: str = "",
+        content: str = "",
+    ):
+        logger.warning(
+            f"The webhook: {self.webhook.name} is currently in test mode"
+        )
+        self.webhook.test_error_message = error_message
+        self.webhook.test_content_type = content_type
+        self.webhook.test_content = content
+        self.webhook.save(
+            update_fields=[
+                "test_content_type",
+                "test_content",
+                "test_error_message",
+            ]
+        )
+
+    def _parse_body(self, content_type: str, body: bytes) -> dict:
+        if content_type == "application/json":
+            data = yaml.safe_load(body.decode())
+        elif content_type == "application/x-www-form-urlencoded":
+            data = urllib.parse.parse_qs(body.decode())
+        else:
+            message = f"Invalid content type passed {content_type}"
+            logger.error(message)
+            raise ParseError(message)
+        return data
+
+    def _embellish_data(self, headers: HttpHeaders, data: dict):
+        data["eda_webhook_name"] = self.webhook.name
+        if not self.webhook.additional_data_headers:
+            return
+        for key in self.webhook.additional_data_headers:
+            value = headers.get(key)
+            if value:
+                data[key] = value
+
+    def _hmac_auth(self, body: bytes, signature: str):
+        hash_object = hmac.new(
+            self.webhook.secret.get_secret_value().encode("utf-8"),
+            msg=body,
+            digestmod=self.webhook.hmac_algorithm,
+        )
+        if self.webhook.hmac_format == "hex":
+            expected_signature = (
+                self.webhook.hmac_signature_prefix + hash_object.hexdigest()
+            )
+        elif self.webhook.hmac_format == "base64":
+            expected_signature = (
+                self.webhook.hmac_signature_prefix
+                + base64.b64encode(hash_object.digest()).decode()
+            )
+
+        if not hmac.compare_digest(expected_signature, signature):
+            message = "Signature mismatch, check your payload and secret"
+            logger.warning(message)
+            if self.webhook.test_mode:
+                self._update_test_data(error_message=message)
+            raise AuthenticationFailed(message)
+
+    def _token_auth(self, token: str):
+        if self.webhook.secret.get_secret_value() != token:
+            message = "Token mismatch, check your token"
+            logger.warning(message)
+            if self.webhook.test_mode:
+                self._update_test_data(error_message=message)
+            raise AuthenticationFailed(message)
+
+    @extend_schema(exclude=True)
+    @action(detail=True, methods=["POST"], rbac_action=None)
+    def post(self, request, *args, **kwargs):
+        try:
+            self.webhook = Webhook.objects.get(uuid=kwargs["pk"])
```

Comment:
Why a parameter? I think it is more usual take the UUID from the uri. Also it would make easier for admins to apply custom logic for specific endpoints in load-balancers.

Reply:
@Alex-Izquierdo I am not sure if the admins would be routing based on each individual webhook, the routing would happen at a higher level from the NGINX configuration which will come from the UI NGINX in this PR https://github.com/ansible/ansible-ui/pull/1738/files